tests/cnf/core/network/metallb/tests/frrk8-tests.go

```diff
@@ -0,0 +1,760 @@
+		externalNad.Definition.Name, []string{fmt.Sprintf("%s/%s", ipv4metalLbIPList[0], "24")})
+
+	By("Creating MetalLb configMap")
+
+	masterConfigMap := createConfigMapWithStaticRoutes(tsparams.LocalBGPASN, ipv4NodeAddrList, hubIPAddresses,
+		externalAdvertisedIPv4Routes, externalAdvertisedIPv6Routes, false, false)
+
+	By("Creating FRR Pod")
+
+	frrPod := createFrrPod(
+		masterNodeList[0].Object.Name, masterConfigMap.Definition.Name, []string{}, staticIPAnnotation)
+
+	return frrPod
+}
+
+func createFrrConfiguration(name, bgpPeerIP string, remoteAS uint32,
+	filteredIP []string, ebgp bool) error {
+	frrConfig := metallb.NewFrrConfigurationBuilder(APIClient, name,
+		NetConfig.MlbOperatorNamespace).WithBGPRouter(tsparams.LocalBGPASN).
+		WithBGPNeighbor(bgpPeerIP, remoteAS, 0)
+	if len(filteredIP) > 0 {
+		frrConfig.WithToReceiveModeFiltered(filteredIP, 0, 0)
+	} else {
+		frrConfig.WithToReceiveModeAll(0, 0)
+	}
+
+	if ebgp {
+		frrConfig.WithEBGPMultiHop(0, 0)
+	}
+
+	frrConfig.
+		WithHoldTime(metav1.Duration{Duration: 90 * time.Second}, 0, 0).
+		WithKeepalive(metav1.Duration{Duration: 30 * time.Second}, 0, 0).
+		WithBGPPassword("bgp-test", 0, 0).
+		WithPort(179, 0, 0)
+
+	_, err := frrConfig.Create()
+
+	if err != nil {
+		glog.V(90).Infof("failed to create a frrconfiugration")
+
+		return err
+	}
+
+	return nil
+}
+
+func createMasterFrrPod(localAS int, ipv4NodeAddrList, hubIPAddresses, externalAdvertisedIPv4Routes,
+	externalAdvertisedIPv6Routes []string, ebgpMultiHop bool) *pod.Builder {
+	masterConfigMap := createConfigMapWithStaticRoutes(localAS, ipv4NodeAddrList, hubIPAddresses,
+		externalAdvertisedIPv4Routes, externalAdvertisedIPv6Routes, ebgpMultiHop, false)
+
+	By("Creating static ip annotation for master FRR pod")
+
+	masterStaticIPAnnotation := pod.StaticIPAnnotation(
+		tsparams.HubMacVlanNADName, []string{"172.16.0.1/24", "2001:100:100::254/64"})
+
+	By("Creating FRR Master Pod")
+
+	frrPod := createFrrPod(
+		masterNodeList[0].Object.Name, masterConfigMap.Definition.Name, []string{}, masterStaticIPAnnotation)
+
+	return frrPod
+}
+
+func createConfigMapWithStaticRoutes(
+	bgpAsn int, nodeAddrList, hubIPAddresses, externalAdvertisedIPv4Routes, externalAdvertisedIPv6Routes []string,
+	enableMultiHop, enableBFD bool) *configmap.Builder {
+	frrBFDConfig := frr.DefineBGPConfigWithStaticRouteAndNetwork(
+		bgpAsn, tsparams.LocalBGPASN, hubIPAddresses, externalAdvertisedIPv4Routes,
+		externalAdvertisedIPv6Routes, removePrefixFromIPList(nodeAddrList), enableMultiHop, enableBFD)
+	configMapData := frr.DefineBaseConfig(tsparams.DaemonsFile, frrBFDConfig, "")
+	masterConfigMap, err := configmap.NewBuilder(APIClient, "frr-master-node-config", tsparams.TestNamespaceName).
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create config map")
+
+	return masterConfigMap
+}
+
+func verifyExternalAdvertisedRoutes(frrPod *pod.Builder, ipv4NodeAddrList, externalExpectedRoutes []string) error {
```

Comment:
Valdate all errors inside the local func via Expect()

Reply:
changed